test/e2e/network_segmentation.go

```diff
@@ -508,6 +509,90 @@ var _ = Describe("Network Segmentation", func() {
 		expectedMessage := fmt.Sprintf("primary network already exist in namespace %q: %q", f.Namespace.Name, primaryNadName)
 		Expect(actualConditions[0].Message).To(Equal(expectedMessage))
 	})
+
+	Context("pod2Egress on a user defined primary network", func() {
+		const (
+			externalContainerName = "ovn-k-egress-test-helper"
+		)
+		var externalIpv4, externalIpv6 string
+		BeforeEach(func() {
+			externalIpv4, externalIpv6 = createClusterExternalContainer(
+				externalContainerName,
+				"registry.k8s.io/e2e-test-images/agnhost:2.45",
+				runExternalContainerCmd(),
+				httpServerContainerCmd(port),
+			)
+
+			DeferCleanup(func() {
+				deleteClusterExternalContainer(externalContainerName)
+			})
+		})
+
+		DescribeTable(
+			"can be accessed to from the pods running in the Kubernetes cluster",
+			func(netConfigParams networkAttachmentConfigParams, clientPodConfig podConfiguration) {
+				netConfig := newNetworkAttachmentConfig(netConfigParams)
+
+				netConfig.namespace = f.Namespace.Name
+				clientPodConfig.namespace = f.Namespace.Name
+
+				By("creating the attachment configuration")
```

Comment:
we should list here the generated network name, so it becomes easier to debug when things go wrong.

Reply:
@maiqueb Do we need a more unified approach (for more than just single test)?  Should we open a follow up issue to track that work?

Reply:
yeah I think all tests need this cleanup

Reply:
Yeah, I guess all tests will benefit from it.